python/pyspark/sql/tests.py

```diff
@@ -476,6 +476,19 @@ def test_udf_defers_judf_initalization(self):
             "judf should be initialized after UDF has been called."
         )
 
+    @unittest.skipIf(sys.version_info < (3, 3), "Unittest < 3.3 doesn't support mocking")
```

Comment:
@holdenk  Do you mean something like `SparkContext._active_spark_context is None`? Then we need to make sure it is tear down if it was initialized after all, right? Isn't mocking cleaner?